cmd/nitro/nitro.go

```diff
@@ -581,3 +596,59 @@ func applyArbitrumAnytrustGoerliTestnetParameters(k *koanf.Koanf) error {
 		"persistent.chain": "goerli-anytrust",
 	}, "."), nil)
 }
+
+func testIndexUpdated(chainDb ethdb.Database, lastBlock uint64) bool {
+	var transactions types.Transactions
+	blockHash := rawdb.ReadCanonicalHash(chainDb, lastBlock)
+	reReadNumber := rawdb.ReadHeaderNumber(chainDb, blockHash)
+	if reReadNumber == nil {
+		return false
+	}
+	for ; ; lastBlock-- {
+		blockHash := rawdb.ReadCanonicalHash(chainDb, lastBlock)
+		block := rawdb.ReadBlock(chainDb, blockHash, lastBlock)
+		transactions = block.Transactions()
+		if len(transactions) == 0 {
+			if lastBlock == 0 {
+				return true
+			}
+			continue
+		}
+		entry := rawdb.ReadTxLookupEntry(chainDb, transactions[len(transactions)-1].Hash())
+		return (entry != nil)
+	}
+}
+
+func testUpdateTxIndex(chainDb ethdb.Database, chainConfig *params.ChainConfig) {
+	lastBlock := chainConfig.ArbitrumChainParams.GenesisBlockNum
+	if lastBlock == 0 {
+		// no Tx, no need to update index
+		return
+	}
+
+	lastBlock -= 1
+	if testIndexUpdated(chainDb, lastBlock) {
+		return
+	}
+
+	log.Info("writing Tx lookup entries")
+	batch := chainDb.NewBatch()
+	for blockNum := uint64(0); blockNum <= lastBlock; blockNum++ {
+		blockHash := rawdb.ReadCanonicalHash(chainDb, blockNum)
+		block := rawdb.ReadBlock(chainDb, blockHash, blockNum)
+		rawdb.WriteTxLookupEntriesByBlock(batch, block)
+		rawdb.WriteHeaderNumber(batch, block.Header().Hash(), blockNum)
+		if (batch.ValueSize() >= ethdb.IdealBatchSize) || blockNum == lastBlock {
```

Comment:
I don't have a recommended change here, I just wanted to say nice find! I had no idea `ethdb.IdealBatchSize` existed :D